app/controllers/nodes_controller.rb

```diff
@@ -22,14 +22,16 @@ def show
   # assigned role, assign a random role to it, and then call the salt
   # orchestration.
   def bootstrap
-    available_roles = [:master]
+    @available_roles = [:master]
```

Comment:
@jimmykarily we cannot `return assigned_roles` because this is a method that responds to a request no ?

Otherwise, a suggestion/nitpick that I have is to move the `[:master]` into a constant in the beginning of the class:

```ruby
INITIAL_AVAILABLE_ROLES = [:master].freeze
```

And then assign the `@available_roles` to it. This is more of a cosmetic change, I know, but I think it reflects better the intention of this first initialization.

That being said, even if it smells ugly, I don't see another way around creating an instance variable honestly 😞